packages/host/app/components/operator-mode/error-display.gts

```diff
@@ -0,0 +1,250 @@
+import { on } from '@ember/modifier';
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+import { dropTask } from 'ember-concurrency';
+
+import perform from 'ember-concurrency/helpers/perform';
+
+import { CopyButton, Button } from '@cardstack/boxel-ui/components';
+import { and, bool, not } from '@cardstack/boxel-ui/helpers';
+import {
+  DropdownArrowDown,
+  DropdownArrowUp,
+  ExclamationTriangleFill,
+} from '@cardstack/boxel-ui/icons';
+
+import SwitchSubmodeCommand from '@cardstack/host/commands/switch-submode';
+import OperatorModeStateService from '@cardstack/host/services/operator-mode-state-service';
+
+import type { FileDef } from 'https://cardstack.com/base/file-api';
+
+import SendErrorToAIAssistant from './send-error-to-ai-assistant';
+
+interface Signature {
+  Element: HTMLElement;
+  Args: {
+    type: 'syntax' | 'runtime';
+    headerText?: string;
+    message?: string;
+    stack?: string;
+    openDetails?: boolean;
+    fileToAttach?: FileDef;
+    viewInCodeMode?: boolean;
+    cardId?: string;
+  };
+}
+
+export default class ErrorDisplay extends Component<Signature> {
+  @tracked private showDetails = this.args.openDetails ?? false;
+
+  @service private declare operatorModeStateService: OperatorModeStateService;
+  @service private declare commandService: any;
+
+  private viewInCodeMode = dropTask(async () => {
+    let switchSubmodeCommand = new SwitchSubmodeCommand(
+      this.commandService.commandContext,
+    );
+    await switchSubmodeCommand.execute({
+      submode: 'code',
+      codePath: `${this.args.cardId}.json`,
+    });
+  });
+
+  private toggleDetails = () => (this.showDetails = !this.showDetails);
+
+  private get errorObject() {
+    return {
+      message: this.args.message ?? '',
+      stack: this.args.stack,
+    };
+  }
+
+  private get headerText() {
+    return this.args.headerText ?? `${this.args.type} Error`;
+  }
+
+  <template>
+    <div class='error-display' data-test-error-display>
+      <div class='error-header'>
+        <div class='error-type' data-test-error-type>
+          <ExclamationTriangleFill class='error-icon' />
+          <span>{{this.headerText}}</span>
+        </div>
+        {{#if
+          (and
+            (bool @fileToAttach)
+            (not this.operatorModeStateService.aiAssistantOpen)
+          )
+        }}
+          <SendErrorToAIAssistant
+            @error={{this.errorObject}}
+            @errorType={{@type}}
+            @fileToAttach={{@fileToAttach}}
+          />
+        {{/if}}
+      </div>
+
+      {{#if @message}}
+        <div class='error-message' data-test-error-message>
+          {{@message}}
+        </div>
+      {{/if}}
+
+      <div class='error-actions'>
+        <CopyButton @textToCopy={{@message}} @width='16px' @heigth='16px' />
+        <button
+          class='toggle-details-button'
```

Comment:
Can probably remove a lot of extra css by using `<Button @kind='text-only' @size='extra-small' ...`